### ISSUE TITLE
[lilliput] destructor must be noexecpt in c++11 (Trac #1666)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1658]
Don't call log_fatal in a destructor.

Warning:

```text
from
    /home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModul
    e.h:4,
from /home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModule.cxx:1:
    /home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModule.cx
    x: In destructor ‘virtual PSSTestModule::~PSSTestModule()’:
/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/I3Logging.h:181:57:
    warning: throw will always call terminate() [-Wterminate]
##__VA_ARGS__) + " (in " + __PRETTY_FUNCTION__ + ")")
^
/home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModule.cxx:17:9:
    note: in expansion of macro ‘log_fatal’
log_fatal("saw ZERO NADA physics frames");
^
/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/I3Logging.h:181:57:
    note: in C++11 destructors default to noexcept
##__VA_ARGS__) + " (in " + __PRETTY_FUNCTION__ + ")")
^
/home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModule.cxx:17:9:
    note: in expansion of macro ‘log_fatal’
log_fatal("saw ZERO NADA physics frames");
^
```

Migrated from https://code.icecube.wisc.edu/ticket/1666
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:10", 
    "description": "Don't call log_fatal in a destructor.\n\nWarning:\n{{{\nfrom\n    /home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModul\n    e.h:4,\nfrom /home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModule.cxx:1:\n    /home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModule.cx\n    x: In destructor \u2018virtual PSSTestModule::~PSSTestModule()\u2019:\n/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/I3Logging.h:181:57:\n    warning: throw will always call terminate() [-Wterminate]\n##__VA_ARGS__) + \" (in \" + __PRETTY_FUNCTION__ + \")\")\n^\n/home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModule.cxx:17:9:\n    note: in expansion of macro \u2018log_fatal\u2019\nlog_fatal(\"saw ZERO NADA physics frames\");\n^\n/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/I3Logging.h:181:57:\n    note: in C++11 destructors default to noexcept\n##__VA_ARGS__) + \" (in \" + __PRETTY_FUNCTION__ + \")\")\n^\n/home/dschultz/Documents/combo/trunk/src/lilliput/private/test/PSSTestModule.cxx:17:9:\n    note: in expansion of macro \u2018log_fatal\u2019\nlog_fatal(\"saw ZERO NADA physics frames\");\n^\n}}}", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067190995086", 
    "component": "combo reconstruction", 
    "summary": "[lilliput] destructor must be noexecpt in c++11", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-04-26T21:12:35", 
    "milestone": "", 
    "owner": "kkrings", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160426T23:35:48: kkrings changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20160621T15:09:57: 
* kkrings changed resolution from "" to "fixed"
* kkrings changed status from "accepted" to "closed"
* kkrings commented:

> In changeset 147651:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="147651"
> Moved log_fatal into Finish method; closes #1658.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:10: 
* olivas commented:

> Milestone July 2016 deleted


* olivas changed milestone from "July 2016" to ""
